cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -218,8 +217,8 @@ impl MockClient {
     }
 }
 
-#[async_trait]
 #[enum_dispatch]
+#[allow(async_fn_in_trait)]
```

Comment:
The compiler complains about some current limitations with async traits, namely that you can't place extra bounds on return types. This would normally be a problem for consumers of the trait, but there are workarounds when necessary and we're the only consumers of the trait.